source/opt/eliminate_dead_input_components_pass.cpp

```diff
@@ -42,6 +43,13 @@ Pass::Status EliminateDeadInputComponentsPass::Process() {
   // Current functionality assumes shader capability.
   if (!context()->get_feature_mgr()->HasCapability(spv::Capability::Shader))
     return Status::SuccessWithoutChange;
+  // Current functionality assumes vert, frag, tesc, tese or geom shader
+  if (stage != spv::ExecutionModel::Vertex &&
```

Comment:
The 2 missing execution models are Kernel & GLCompute.
Because you check for Shader capability, Kernel is not possible.
So here you only disallow compute stage, maybe flip the condition and add a comment with a TODO(4988) saying compute should be OK, but needs to be checked.

Reply:
There are many other models that are in the Shader category, for example, mesh, task and all the ray tracing shaders which are currently not supported.

My preference is to be very explicit about which shaders are currently supported. I think it is clearer and safer.

Reply:
Oh, right, looked at the enum and forgot about those.
Yes, you are right, would be more future-proof to explicitly allow!